data-access-layer/eventmesh/LockManager.js

```diff
@@ -167,28 +172,45 @@ class LockManager {
   */
   /**
    * @param {string} resourceId - Id (name) of the resource that is being locked. In case of deployment lock, it is instance_id
-   * @param {number} [maxRetryCount=CONST.ETCD.MAX_RETRY_UNLOCK] - Max unlock attempts
+   * @param {string} lockId - lockId which is return by lockManager.lock when lock is acquired
+   * @param {number} [maxRetryCount=CONST.APISERVER.MAX_RETRY_UNLOCK] - Max unlock attempts
    */
 
-  unlock(resourceId, maxRetryCount, retryDelay) {
+  unlock(resourceId, lockId, maxRetryCount, retryDelay) {
     assert.ok(resourceId, `Parameter 'resourceId' is required to release lock`);
-    maxRetryCount = maxRetryCount || CONST.ETCD.MAX_RETRY_UNLOCK;
+    // TODO-PR: making lockId not mendatory as currently we don't have deployment resources
+    // hence from lastOperation call we can't pass lockId for unlock call
+    // assert.ok(lockId, `Parameter 'lockId' is required to release lock`);
+    maxRetryCount = maxRetryCount || CONST.APISERVER.MAX_RETRY_UNLOCK;
     retryDelay = retryDelay || CONST.APISERVER.RETRY_DELAY;
     logger.debug(`Attempting to unlock resource ${resourceId}`);
     return utils.retry(tries => {
       logger.info(`+-> Attempt ${tries + 1} to unlock resource ${resourceId}`);
-      return eventmesh.apiServerClient.deleteResource({
-          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.LOCK,
-          resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT_LOCKS,
-          resourceId: resourceId
-        })
+      const opts = {
+        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.LOCK,
+        resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT_LOCKS,
+        resourceId: resourceId,
+        status: {
+          state: CONST.APISERVER.RESOURCE_STATE.UNLOCKED
+        }
+      };
+      if (lockId) {
+        opts.metadata = {
+          resourceVersion: lockId
+        };
+      }
+      return eventmesh.apiServerClient.updateResource(opts)
         .tap(() => logger.debug(`Successfully unlocked resource ${resourceId} `))
         .catch(err => {
+          if (err instanceof Conflict) {
```

Comment:
can leverage `.catch(conflict, (err)=> {})` instead of the above syntax for consistency across code base.